cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java

```diff
@@ -125,7 +127,11 @@ public static void addNamedOutput(Job job, String namedOutput, String outputForm
     conf.set(MO_PREFIX + namedOutput + FORMAT, outputFormatClass);
     conf.setClass(MO_PREFIX + namedOutput + KEY, keyClass, Object.class);
     conf.setClass(MO_PREFIX + namedOutput + VALUE, valueClass, Object.class);
-    ConfigurationUtil.setNamedConfigurations(conf, namedOutput, outputConfigs);
+    ConfigurationUtil.setNamedConfigurations(conf, computePrefixName(namedOutput), outputConfigs);
+  }
+
+  private static String computePrefixName(String outputName) {
+    return PREFIXED_CONF_PREFIX + outputName + CONF;
   }
```

Comment:
That assumes that output names can't contain a period? Definitely not the string .conf